src/main/java/com/faforever/client/game/GameService.java

```diff
@@ -609,11 +613,25 @@ private CompletableFuture<Void> startGame(GameLaunchResponse gameLaunchMessage)
         .thenApply(adapterPort -> {
           fafServerAccessor.setPingIntervalSeconds(5);
           gameKilled = false;
-          try {
-            process = forgedAllianceService.startGameOnline(gameLaunchMessage,
-                adapterPort, localReplayPort, rehostRequested);
-          } catch (IOException e) {
-            throw new GameLaunchException("Could not start game", e, "game.start.couldNotStart");
+          if (isMatchmakerGame) {
+            leaderboardService.getActiveLeagueEntryForPlayer(playerService.getCurrentPlayer(), gameLaunchMessage.getLeaderboard()).thenAccept(leagueBeanOptional -> {
+              String divisionName = leagueBeanOptional.map(bean ->
+                      bean.getSubdivision().getDivision().getNameKey() + " " + bean.getSubdivision().getNameKey())
+                  .orElse("unlisted");
+              try {
+                process = forgedAllianceService.startGameOnlineWithDivision(gameLaunchMessage,
+                    adapterPort, localReplayPort, rehostRequested, divisionName);
+              } catch (IOException e) {
+                throw new GameLaunchException("Could not start game", e, "game.start.couldNotStart");
+              }
+            });
+          } else {
```

Comment:
Since this is only specific to matchmaker it should go in the startMatchmaker function not the generic start game function